src/BuildExecutable.jl

```diff
@@ -261,7 +261,7 @@ function emit_cmain(cfile, exename, relocation)
 
         int main(int argc, char *argv[])
         {
-            char sysji[] = "$(sysji).ji";
+            char sysji[] = "$(sysji).$(Libdl.dlext)";
```

Comment:
this will make backtraces worse on Windows with Julia 0.4, may want to make this version 0.5-only?